pylint/pyreverse/dot_printer.py

```diff
@@ -103,31 +102,11 @@ def _build_label_for_node(
         # Add class methods
         methods: List[astroid.FunctionDef] = properties.methods or []
         for func in methods:
-            return_type = (
-                f": {get_annotation_label(func.returns)}" if func.returns else ""
-            )
-
-            if func.args.args:
-                arguments: List[astroid.AssignName] = [
-                    arg for arg in func.args.args if arg.name != "self"
-                ]
-            else:
-                arguments = []
-
-            annotations = dict(zip(arguments, func.args.annotations[1:]))
-            for arg in arguments:
-                annotation_label = ""
-                ann = annotations.get(arg)
-                if ann:
-                    annotation_label = get_annotation_label(ann)
-                annotations[arg] = annotation_label
-
-            args = ", ".join(
-                f"{arg.name}: {ann}" if ann else f"{arg.name}"
-                for arg, ann in annotations.items()
-            )
-
-            label += fr"{func.name}({args}){return_type}\l"
```

Comment:
The same logic is used by ``PlantUmlPrinter``, so I extracted a method and put it inside the base ``Printer`` class.